### ISSUE TITLE
train_on_inputs 

### ISSUE THREAD

**Issue by xpang-sf** [tloen/alpaca-lora#499]
May I ask a simple question:  what is train_on_inputs  used for? by default, the flaf=True. I don't see original code (https://github.com/tatsu-lab/stanford_alpaca) has this flag. thanks a lot and look forward to hearing from you.

**Comment by xpang-sf** [comment]
May I ask: when run fine-tuning, the setting should be: train_on_inputs: bool = False , right? why train_on_inputs: bool = True  in the code?

**Comment by freQuensy23-coder** [comment]
Llm used to predict next token (word) based on previous history. To train such model we should give it a lot of texts, it will be divided into tokens (words). Model will try to predict each next word, loss func will be calculated and weights will be changed.  (for ex. for 'New York is capital of the USA, isn't it?  RESPONSE: No it does not'  predict  York based by New, then predict is based by 'New York' then biggest by the 'New York is' ..... etc). 

hen training on inputs is set to false model will not train on input. So while training it will predict No based on 'New York is capital of the USA, isn't it?  RESPONSE:' then it based on 'New York is capital of the USA, isn't it?  RESPONSE: No' etc

This script mask input with default mask value -100 using [this](https://github.com/tloen/alpaca-lora/blob/8bb8579e403dc78e37fe81ffbb253c413007323f/finetune.py#L155C9-L171C44) code

```
if not train_on_inputs:
            user_prompt = prompter.generate_prompt(
                data_point["instruction"], data_point["input"]
            )
            tokenized_user_prompt = tokenize(
                user_prompt, add_eos_token=add_eos_token
            )
            user_prompt_len = len(tokenized_user_prompt["input_ids"])

            if add_eos_token:
                user_prompt_len -= 1

            tokenized_full_prompt["labels"] = [
                -100
            ] * user_prompt_len + tokenized_full_prompt["labels"][
                user_prompt_len:
            ]  # could be sped up, probably
```
